store.tsx

```diff
@@ -1,12 +1,23 @@
 import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
 import { productsApi } from "./services/productsApi";
 import { cartReducer } from "./redux/cartSlice";
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { persistStore, persistReducer } from 'redux-persist'
 
 import { combineReducers } from "@reduxjs/toolkit";
 
-const allReducers = combineReducers({
-  [productsApi.reducerPath]: productsApi.reducer,
+const persistConfig = {
+  key: "root",
+  version: 1,
+  storage: AsyncStorage,
+}
+const rootReducer = combineReducers({
   cart: cartReducer,
+})
+const persistedReducer = persistReducer(persistConfig, rootReducer)
+
+const allReducers = combineReducers({
+  [productsApi.reducerPath]: productsApi.reducer, persistedReducer,
```

Comment:
Nice job getting this figured out!  So, the guide you used is likely out of date considering that you're using Redux Toolkit.  Here's a Redux Toolkit version of the above that should work and be easier to interact with:

```
const persistedCartReducer = persistReducer(persistConfig, cartReducer);

export const store = configureStore({
  reducer: {
    cart: persistedCartReducer,
    [productsApi.reducerPath]: productsApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(productsApi.middleware),
});
```

You don't need to combine the reducers now.  That's something we used to have to do when Redux only had a single reducer entry point.  Hope this helps reduce the complexity of your app!

Reply:
Thanks so much Kendrick for your feedback I will make the changes accordingly !